homeassistant/components/upnp.py

```diff
@@ -36,17 +58,22 @@ def setup(hass, config):
         _LOGGER.exception("Error when attempting to discover an UPnP IGD")
         return False
 
-    base_url = urlsplit(hass.config.api.base_url)
-    host = base_url.hostname
-    external_port = internal_port = base_url.port
+    unit = config[DOMAIN].get(CONF_UNITS)
+    discovery.load_platform(hass, 'sensor', DOMAIN, {'unit': unit}, config)
```

Comment:
You cannot call sync methods inside a coroutine.

I would suggest you change it back to `setup` and not make it async.